actions/chats.js

```diff
@@ -139,7 +160,27 @@ export const fetchCreateChat = (name) => {
             .then((data) => {
                 dispatch(createChat(data));
             })
-            .catch((error) => console.log('Error:', error));
+            .catch((error) => {
+                console.log('Error:', error);
+                dispatch(setError(error));
+            });
+    };
+};
+
+export const fetchFindUserByEmail = (email) => {
+    return async (dispatch) => {
+        const headers = await addAuthHeader();
+        return fetch(`${baseUrl}user/${email}`, {
+            headers
+        })
+            .then(handleResponse)
+            .then((data) => {
+                dispatch(setFoundUser(data));
+            })
+            .catch((error) => {
+                console.log('Error:', error);
+                dispatch(setError(error));
+            });
```

Comment:
Transferred the fetch GET request to `/user/:email` to the actions (i.e., this file as opposed to in the component file).